backend/src/game/Dealer.ts

```diff
@@ -46,6 +50,38 @@ export class Dealer {
     });
   }
 
+  public static foldCards(pokerTable: PokerTable, userId: number) {
+    const game = pokerTable.getGame();
+    if (!game) {
+      return Result.error(new PokerTableDoesNotExistError());
+    }
+
+    const seats = pokerTable.getSeats();
+    for (const seat of seats) {
+      if (seat.getPlayer()?.getUserId() === userId && game.getGameState().getSeatToAct() === seat.getSeatNumber()) {
+        const player = seat.getPlayer();
+        const playerCards = player?.getCards();
+        if (!(playerCards && playerCards.length > 0)){
+          return Result.error(new PlayerAlreadyFolded());
+        }
+
+        if (!player){
+          return Result.error(new PlayerNotFoundAtPokerTableError());
+        }
+          
+        player.foldCards();
+        GameEmitter.eventEmitter.emit('foldCards', pokerTable.getName(), player.getUsername(), seat.getSeatNumber());
+        if (pokerTable.playersRemaining()){
+          Dealer.updateTurn(pokerTable);
+          return Result.success();
+        }
+
+        // End game logic here
+      }
+    }
+    return Result.error(new PlayerNotFoundAtPokerTableError());
```

Comment:
There are two errors here: PlayerNotFound and NotPlayersTurn

Reply:
@tommy-trombone in response to the unit tests sporadically failing, you're on the money. 

You can test this by logging the response to `Dealer.foldCards` as it returns an error, however because you aren't returning the `NotPlayersTurn` error, you will see a `PlayerNotFound` error which is incorrect as per above. 

So two things to do: 
1. Add the new error and return it properly
2. The unit test should instead get the player at the seat to act: `game.getGameState().getSeatToAct()` and fold that player instead.

Reply:
Yeah good points. 

The unit test is overly hard coded. I'll update.